core/arch/atomic_exports.h

```diff
@@ -932,15 +935,15 @@ atomic_compare_exchange_int(volatile int *var, int compare, int exchange)
     ASSERT(ALIGNED(var, 4));
     /* This code is based on GCC's atomic_compare_exchange_strong().
      * The fence ensures no re-ordering beyond the start of the lr/sc loop.
-     * The lr/sc use acquire ordering to ensure that the full loop has
-     * sequentially consistent memory order semantics.
+     * The lr/sc use acquire and release ordering to ensure that the full loop
+     * has sequentially consistent memory order semantics.
```

Comment:
We only need acquire/release: we don't need sequential consistency.

Reply:
Fixed in https://github.com/DynamoRIO/dynamorio/pull/6089/commits/43eaccfa161c78355d61b04bbcd317e3466edebd.